pyiron_atomistics/atomistics/structure/analyse.py

```diff
@@ -47,6 +76,234 @@ def get_average_of_unique_labels(labels, values):
         return mean_values.flatten()
     return mean_values
 
+class Interstitials:
+    """
+    Class to analyse interstitial sites
+
+    This class internally does the following steps:
+
+        1. Initialize grid points (or Voronoi vertices) which are considered as
+            interstitial site candidates.
+        2. Eliminate points within a distance from the nearest neighboring atoms as
+            given by `min_distance`
+        3. Initialize neighbor environment using `get_neighbors`
+        4. Shift interstitial candidates to the nearest symmetric points with respect to the
+            neighboring atom sites/vertices.
+        5. Kick out points with large neighbor distance variances
+        6. Cluster interstitial candidates to avoir point overlapping.
+
+    The interstitial sites can be obtained through `get_interstitials`
+
+    In complex structures (i.e. grain boundary, dislocation etc.), the default parameters
+    should be chosen properly. In order to see other quantities, which potentially
+    characterize interstitial sites, see the following functions from
+    `structure.analyse.interstitials`:
+
+        - `get_variance()`
+        - `get_distance()`
+        - `get_steinhardt_parameter()`
+        - `get_volume()`
+        - `get_area()`
+    """
+    def __init__(self, structure, n_gridpoints_per_angstrom=5, min_distance=1, use_voronoi=False):
+        """
+        Args:
+            structure (pyiron_atomistics.atomistics.structure.atoms.Atoms): Reference structure
+            n_gridpoints_per_angstrom (int): Number of grid points per angstrom for the
+                initialization of the interstitial candidates. The finer the mesh (i.e. the larger
+                the value), the likelier it is to find the correct sites but then also it becomes
+                computationally more expensive. Ignored if `use_voronoi` is set to `True`
+            min_distance (float): Minimum distance from the nearest neighboring atoms to the
+                positions for them to be considered as interstitial site candidates.
+            use_voronoi (bool): Use Voronoi vertices for the initial interstitial candidate
+                positions instead of grid points.
+        """
+        self.min_distance = min_distance
+        self.structure = structure
+        if use_voronoi:
+            self.positions = structure.analyse.get_voronoi_vertices()
+        else:
+            self.positions = self._create_gridpoints(
+                n_gridpoints_per_angstrom=n_gridpoints_per_angstrom
+            )
+        if min_distance > 0:
+            self._remove_too_close()
+
+    @property
+    def num_neighbors(self):
+        return self._num_neighbors
+
+    @num_neighbors.setter
+    def num_neighbors(self, n):
+        self.reset()
+        self._num_neighbors = n
+
+    def reset(self):
+        self._hull = None
+        self._neigh = None
+
+    @property
+    def neigh(self):
```

Comment:
Short docstring for the properties would be helpful, like "neigh" is the neighborhood of interstitial candidate site, "positions" are their positions, not atomic ones, etc.  It makes sense once you've seen/used the code, but probably not the first time round.